File: 연습/k2022/spring/연습/sns/karma/src/main/java/com/sns/karma/exception/CustomException.java
package com.sns.karma.exception;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class CustomException extends RuntimeException{
    private ErrorCode errorCode;
    private String message = null;


    @Override
    public String getMessage() {
        if (message == null) {
            return errorCode.getMessage();
        } else {
            return String.format("Error Code : %s \n Error message : %s", errorCode.getMessage(), message);
        }
    }
}

File: lib/core/response/custom_exception.dart
import '../constant/constant.dart';

class CustomException implements Exception {
  final String message;
  final ErrorCode code;

  CustomException(this.code, {this.message = 'custom exception'});

  @override
  String toString() => 'CustomException: $message';

  factory CustomException.from(Exception exception) {
    // 에러 유형별 코드 세분화하기
    return CustomException(ErrorCode.unknownError,
        message: exception.toString());
  }
}
